astropy/timeseries/periodograms/lombscargle/implementations/utils.py

```diff
@@ -12,6 +12,29 @@ def bitceil(N):
     return 1 << int(N - 1).bit_length()
 
 
+def next_fast_len(N):
+    """
+    Find the next number greater, than N, which can be expressed
+    as 2^n * 3^m for integer values of n and m and m <= 4.
+    """
+    Nfft = bitceil(N)
+
+    if (Nfft // 16) * 9 > N:
+        Nfft //= 16
+        Nfft *= 9
+    elif (Nfft // 128) * 81 > N:
+        Nfft //= 128
+        Nfft *= 81
+    elif (Nfft // 4) * 3 > N:
+        Nfft //= 4
+        Nfft *= 3
+    elif (Nfft // 32) * 27 > N:
+        Nfft //= 32
+        Nfft *= 27
+
+    return Nfft
```

Comment:
I'm confused. Is it expected that this function returns `bitceil(N)` if none of the conditions are met ? This doesn't seem to be implied by the docstring. I could see this resolved by either updating the docstring, or changing the code, but really I don't presume of your intention here.

Reply:
> Also I've just realised, that if the grid begins at zero the np.fft.ifft could be replaced with np.fft.rfft (in the 'fasper' branch only) making the execution noticeably faster and keeping the method mathematically equivalent to the current version.

I would recommend trying this out as a follow up PR so as to make this one easier to review